### PR TITLE
feat: add `average` function

### LINKED ISSUE #77: Add `average` function
# Suggestion

## 💻 Use Cases

```ts
average([1,2,3,4,5]); // 3
await average(toAsync([1,2,3,4,5])); // 3
```

### PR DESCRIPTION
**Pull request by ppeeou**
Fixes #77

### CODE REVIEW

File: src/average.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  iterable: T,
+): ReturnValueType<T> {
+  let size = 0;
+
+  if (isIterable(iterable)) {
+    return pipe(
+      iterable as Iterable<number>,
+      peek(() => size++),
+      sum,
+      (a) => (size === 0 ? NaN : a / size),
```
Comment (hg-pyun) [inline src/average.ts:32]: I think this part is the same as the logic below. Is there any way to combine it into one?

**Review by hg-pyun — APPROVED** [review]
It's a minor comment, so I approved.

File: src/average.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { peek, pipe, sum } from ".";
```
Comment (shine1594) [inline src/average.ts:1]:
```suggestion
import peek from "./peek";
import pipe from "./pipe";
import sum from "./sum";
```

File: src/average.ts (hunk 1 above)
Comment (ppeeou) [reply src/average.ts:34]: Do you want to create it as a function?
like this..? `const divide = (a: number) => size === 0 ? NaN : a / size;`

Comment (shine1594) [reply src/average.ts:34]: I guess that @hg-pyun says something like this:
```ts
if (isIterable(iterable) || isAsyncIterable(iterable)) {
  return pipe(
    iterable as Iterable<number> | AsyncIterable<number>,
    peek(() => size++),
    sum,
    (a) => (size === 0 ? NaN : a / size),
  ) as ReturnValueType<T>;
}
```

**Review by shine1594 — APPROVED** [review]

File: src/average.ts (hunk 1 above)
Comment (ppeeou) [reply src/average.ts:34]: We can also combine after this work
I think we can merge it first and then edit everything later.

**Pull request merged**